src/servers/src/http/handler.rs

```diff
@@ -62,6 +62,8 @@ pub struct SqlQuery {
     // specified time precision. Maybe greptimedb format can support this
     // param too.
     pub epoch: Option<String>,
+    // request source, ["dashboard"]
```

Comment:
question: how to identify the requests are from dashboard?

Reply:
Obviously, we must modify the dashboard request body, add a new arg `source`. Maybe https://github.com/GreptimeTeam/greptimedb/issues/3716#issuecomment-2105590075 more reasonable？

Reply:
I thought there was an identifier in the request body before.

If there is no identifier in the original request(which means we need to modify the dashboard anyway), my advice of adding the parameters for limiting query results may be reasonable. 

The implementation may be more straightforward than the current PR. For example:

- Add `limit` parameters:

  ```
  curl -X POST \
    -H 'Authorization: Basic {{authentication}}' \
    -H 'Content-Type: application/x-www-form-urlencoded' \
    -d 'sql={{SQL-statement}}' \
    http://{{API-host}}/v1/sql?db={{db-name}}&limit=1000
  ```

- Parse the parameter and handle the query(most of the logic are same as the current PR).

@realtaobo @MichaelScofield

Reply:
> If there is no identifier in the original request(which means we need to modify the dashboard anyway), my advice of adding the parameters for limiting query results may be reasonable.

Sure, this is a design decision. It hinges on whether future projects will leverage the request source for additional tasks, like data statistics. Both approaches are valid, so the optimal choice depends on project's development roadmap.

Reply:
A `limit` parameter is reasonable. The dashboard could also add an identifier in its request as it might be useful.

Reply:
Ok, add a "limit" parameter is fine by me. However, I suggest adding a "total_rows" field in the response. Otherwise user could be misguided.

Reply:
> I suggest adding a "total_rows" field in the response. Otherwise user could be misguided.

@MichaelScofield  "total_rows" field has been added. I will add a new parameter limit later.